src/platform/Linux/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -123,96 +170,247 @@ void LinuxWiFiDriver::Shutdown()
+    outNetworkIndex   = 0;
+    bool foundNetwork = false;
+
+    VerifyOrReturnError(mCurrentNetId != 0, Status::kBoundsExceeded);
+
+    for (uint8_t networkIndex = 0; networkIndex < kMaxNetworks; networkIndex++)
+    {
+        if (mStagingNetworks[networkIndex].Matches(networkId))
+        {
+            foundNetwork    = true;
+            outNetworkIndex = networkIndex;
+            if (networkIndex == mStagedConnectedNetworkIndex)
+            {
+                mStagedConnectedNetworkIndex = -1;
+            }
+            break;
+        }
+    }
+
+    VerifyOrReturnError(foundNetwork, Status::kNetworkIDNotFound);
 
     // Use empty ssid for representing invalid network
-    mStagingNetwork.ssidLen = 0;
+    mStagingNetworks[outNetworkIndex].ssidLen = 0;
+
+    // Will need to reorder the list.
+    // Shift networks that are at a lower priority than the network to be removed if present in the list.
+    ShiftNetworkAfterRemove();
+    mCurrentNetId--;
+
     return Status::kSuccess;
 }
 
+bool LinuxWiFiDriver::StartReorderingEntries(uint8_t index, int8_t foundNetworkAtIndex)
+{
+    WiFiNetwork tempNetwork[kMaxNetworks];
+    int count      = 0;
+    int indexCount = 0;
+
+    if (foundNetworkAtIndex < 0 || index == foundNetworkAtIndex)
+    {
+        return false;
+    }
+
+    // Directly assigning the network to the new position
+    tempNetwork[index] = mStagingNetworks[foundNetworkAtIndex];
+
+    for (uint8_t networkIndex = 0; networkIndex < kMaxNetworks; networkIndex++)
+    {
+        if (indexCount < kMaxNetworks && index != indexCount)
+        {
+            if (count == foundNetworkAtIndex)
+            {
+                count++;
+            }
+
+            tempNetwork[indexCount] = mStagingNetworks[count];
+
+            if (count != foundNetworkAtIndex)
+            {
+                count++;
+            }
+        }
+
+        if (count > (kMaxNetworks - 1))
+        {
+            count = 0;
+        }
+        indexCount++;
+    }
+
+    memcpy(mStagingNetworks, tempNetwork, sizeof(mStagingNetworks));
+
+    return true;
+}
+
 Status LinuxWiFiDriver::ReorderNetwork(ByteSpan networkId, uint8_t index, MutableCharSpan & outDebugText)
 {
+    int8_t foundNetworkAtIndex = -1;
     outDebugText.reduce_size(0);
-    VerifyOrReturnError(mStagingNetwork.Matches(networkId), Status::kNetworkIDNotFound);
-    VerifyOrReturnError(index == 0, Status::kOutOfRange);
-    // We only support one network, so reorder is actually no-op.
+
+    // Check if network does not contain a matching entry
+    for (uint8_t networkIndex = 0; networkIndex < kMaxNetworks; networkIndex++)
+    {
+        if (mStagingNetworks[networkIndex].Matches(networkId))
+        {
+            foundNetworkAtIndex = networkIndex;
+            break;
+        }
+    }
+    VerifyOrReturnError(foundNetworkAtIndex != -1, Status::kNetworkIDNotFound);
+
+    // Network index larger to the current number of entries return out of range
```

Comment:
What does "larger to" mean?

Reply:
Update the comment `Return out of range if the index value is greater than mStagingNetworkCount i.e the number of enties in mStagingNetworks`